Explore an alternative way of explaining states (#11512)
readthedocs/readthedocs.org: docs/user/versions.rst

--------------

Each version of your documentation has a combination of three states (**Active**, **Public**, and **Hidden**) which determine its visibility on your site:

You can change the states for each version of your documentation in the :guilabel:`Versions` tab of your project.

**Active** or **Inactive**
  - **Active** docs are visible, and builds can be triggered for the documentation.
  - Docs for **Inactive** versions *are deleted* and builds cannot be triggered.

**Hidden** or **Not hidden**
  - **Hidden** docs are not listed on the :term:`flyout menu` on the docs site,
    or shown in search results from another version on the docs site
    (like on search results from a superproject).
  - **Not hidden** docs are listed on the :term:`flyout menu` on the docs site,
    and are shown in search results on the docs site.

  Hiding a version doesn't make it private,
  any user with a link to its docs can still see it.
  This is useful when:

  - You no longer support a version, but you don't want to remove its docs.
  - You have a work in progress version and don't want to publish its docs just yet.

  Hidden verions are listed as ``Disallow: /path/to/version/``

>   in the default `robots.txt file <https://www.robotstxt.org/>`__ created by Read the Docs.

Reviewer: We should probably link to our robots documentation page here instead: https://docs.readthedocs.io/en/stable/reference/robots.html

they will be redirected to the **Default version**.
This defaults to **latest**,
but could also point to your latest released version.

Versioning workflows
--------------------

Read the Docs makes certain assumptions about your documentation version defaults,

> all of which can be reconfigured if you need to:

Author: ```suggestion
all of which can be reconfigured if necessary:
```

- ``latest`` version points to the most up to date development code.
  If you develop on a branch that is different than the default for your version control system,
  set the **Default Branch** to the branch you use.

- **tags** are semantic versioning compatible (according to  `PEP 440`_) snapshots

>   of your documentation. The most recent semantic tag maps to the ``stable`` tag.

Reviewer: > of your documentation. The most recent semantic tag maps to the ``stable`` tag.

I think this should say "of your documentation. The most recent semantic tag maps to the ``stable`` version". Right?

Author: yep, fix incoming.

Author: while we're here. What is more accurate than "most recent semantic tag"? Something that implies the highest stage version number, not the most recent in time?

Reviewer: Maybe "the highest semantic version", 🤔 ? I'm not really good at English 😅

.. note::
    We in fact are parsing your tag names against the rules given by
    `PEP 440`_. This spec allows "normal" version numbers like ``1.4.2`` as
    well as pre-releases. An alpha version or a release candidate are examples
    of pre-releases and they look like this: ``2.0a1``.

    We only consider non pre-releases for the ``stable`` version of your
    documentation.

If you have documentation changes on a **long-lived branch**,
you can build those too.
This will allow you to see how the new docs will be built in this branch of the code.
Generally you won't have more than 1 active branch over a long period of time.
The main exception here would be **release branches**,
which are branches that are maintained over time for a specific release number.

.. _PEP 440: https://www.python.org/dev/peps/pep-0440/

Logging out
-----------

When you log in to a documentation site, you will be logged in until you close your browser.
To log out, click on the :guilabel:`Log out` link in your documentation's :term:`flyout menu`.
This is usually located in the bottom right or bottom left, depending on the theme design.
This will log you out from the current domain,

> but not end any other session that you have active.

Reviewer: This is not available anymore. Currently, there is no "Log out" button. We should probably remove this.

This defaults to **latest**,
but could also point to your latest released version.

Versioning workflows
--------------------

> RTD makes certain assumptions about your documentation version defaults,

Reviewer 2: I don't think we ever use a shortened version like this.

```suggestion
Read the Docs makes certain assumptions about your documentation version defaults,
```

all of which can be reconfigured if you need to:

- ``latest`` version points to the most up to date development code.
  If you develop on a branch that is different than the default for your version control system,
  set the **Default Branch** to the branch you use.

- **tags** are semantic versioning compatible (according to  `PEP 440`_) snapshots
  of your documentation. The most recent semantic tag maps to the ``stable`` tag.

  Semantic versioning allows "normal" version numbers like ``1.4.2``, as
  well as pre-releases like this: ``2.0a1``. The ``stable`` version of your documentation never includes a pre-release.

- If you have documentation changes on a **long-lived branch**,
  you can build those too, to see how the new docs will be built.

>   Generally you won't have more than 1 active branch over a long period of time,
>   apart from **release branches**, maintained over time for a specific release.

Reviewer 2: ```suggestion
 This is most useful for **release branches**, which are maintained over time for a specific release. An example would be a ``2.1`` branch that is kept up to date with the latest ``2.1.x`` release. 
```
